source/Controlled.tsx

```diff
@@ -517,12 +529,14 @@ class ControlledBase extends Component<ControlledPropsWithDefaults, ControlledSt
     window.addEventListener('touchend', this.handleTouchMove, { passive: true })
     window.addEventListener('touchcancel', this.handleTouchCancel, { passive: true })
 
-    this.refModalImg.current?.addEventListener?.('transitionend', () => {
-      setTimeout(() => {
-        this.setState({ modalState: ModalState.LOADED })
-        window.addEventListener('resize', this.handleResize, { passive: true })
-      }, 0)
-    }, { once: true })
+    this.refModalImg.current?.addEventListener?.('transitionend', this.handleZoomEnd, { once: true })
+  }
+
+  handleZoomEnd = () => {
+    setTimeout(() => {
+      this.setState({ modalState: ModalState.LOADED })
+      window.addEventListener('resize', this.handleResize, { passive: true })
+    }, 0)
```

Comment:
This other set of changes is just a bit of paranoia to make sure we clear out our listeners on unmount in case anything goes wonky.